### ISSUE TITLE
BO - Saving product - Error not clear- the error gets eaten

### ISSUE THREAD

**Issue by tswfi** [PrestaShop/PrestaShop#25691]
#### Describe the bug

When saving a product in BO if there is a module that throws an exception (or some other exception happens) the error is eaten and there is no easy way to check what happened (nothing in logs, debug mode does not help).

#### Expected behavior

Any error while saving the product should show an error message of why saving failed.

#### Steps to Reproduce

1. Go to BO product edit
2. In the Combinations Tab > Edit a combination
3. Set a reference with this expression `aaaaaaaaaaaaaaaaaaaassssssssssssssssssssssssssssssssssssssssssssssssssssssssccccccccccccccccccccccccccccccccccccccc`
2. Click on save
3. see "Unable to update settings" no other indication
4. check the network response, the response is just `[]`

#### Reason

ProductController catches exceptions here https://github.com/PrestaShop/PrestaShop/blob/643df036aa53744b9a5e409cb363e5c7e94559b2/src/PrestaShopBundle/Controller/Admin/ProductController.php#L605 and responds only with `[]`.

#### One possible solution

```php
        } catch (Exception $e) {
            // this controller can be called as an AJAX JSON route or a HTML page
            // so we need to return the right type of response if an exception it thrown
            if ($request->isXmlHttpRequest()) {
                return $this->returnErrorJsonResponse(
                    ['step1_name_1' => [$e->getMessage()]],
                    Response::HTTP_INTERNAL_SERVER_ERROR
                );
            }

            throw $e;
        }
```

This will show the error message under the product name field.

I can make a PR but it might be better to have someone from UX to first tell how this kind of errors should be shown. (and it probably requires some JS magic also)

#### Additional information

* PrestaShop version: develop branch
* PHP version: whatever comes when docker-compose up is used in develop branch right now

**Comment by florine2623** [comment]
Hello @tswfi ,

Can you check if your issue is related to https://github.com/PrestaShop/PrestaShop/issues/20279 ?

Thanks!

**Comment by tswfi** [comment]
@florine2623 #20279 is more generic and seems to affect other pages. AFAIK the current dev branch gives quite good error messages on the product edit page as long as the error is something that was "expected" by the ProductController (form checks etc).

But if a module (or probably even database connection error) happens the user will only get a generic "Unable to update.. " notification and no way (even from logs) to see what failed.

Eating the exception in a way that gives out nothing makes debugging stuff really hard (yeah, I used way too many hours finding  why one product would not save today :)

Also as the exception might happen quite late in the process so some data might get saved correctly as the save is not atomic (but that is another issue :)

**Comment by khouloudbelguith** [comment]
Hi @tswfi,

I manage to reproduce the issue with PS1777 and PS1769 and 178x branch (Product V1).
Steps to reproduce the issue:
1. Go to BO > Catalog > Edit a product with combinations
2. In the Combinations Tab > Edit a combination
3. Set a reference with this expression `aaaaaaaaaaaaaaaaaaaassssssssssssssssssssssssssssssssssssssssssssssssssssssssccccccccccccccccccccccccccccccccccccccc`
4. Click on Save
5. Check the console 
![image](https://user-images.githubusercontent.com/16067358/131126741-2dea02bf-66b7-4652-925a-bf0b188d65ba.png)

I'll add this to the debug backlog so that it's fixed.

Please be aware that due to the high number of reported issues, some bugs might take a very long time to be resolved. If this issue is important to you and you cannot wait for it to be fixed on the project's own time, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you. 

If you manage to get the problem fixed, then please contribute the fix back to the project as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/). Remember that the more people contribute, the better PrestaShop becomes for everyone.

Thanks!

**Comment by tswfi** [comment]
@khouloudbelguith yeah, that is actually a different problem :D The form does not check the field value correctly and lets the save happen which then errors out later down the save methods and the error is eaten.

Anyways, I created a PR for this so that the user can get at least some identification on what went wrong. #25734

**Comment by MatShir** [comment]
It should be fixed on 8.1 :)
